### PR TITLE
Some tracking lane adjustments

### PR DESCRIPTION
**Pull request by sanlil**
This is some adjustments I had to do to the tracking lane logic before adding the icons to the tracking lane json. Let me know if there is something I have misunderstood and please test it on Android since I have only tested it on iOS.

* Make sure that the tracking settings view doesn't crash if a row is removed from the json file
* Rename some variables related to tracking lanes:
`trackingLaneDefinition` (the json file)
`laneCategories` (Observations, Symptoms, Activities)
`laneTitles` (e.g. Temperature, Cervical Mucus, Bleeding...)
`optionName` (e.g. Light, Medium, Heavy...)
* Add button on home screen to clear all stored data
* Load tracking lanes when tracking view is focused since updates from the settings weren't visible if the view had been previously loaded
* Fix so that option gets tracked on press (maybe this was an iOS issue only, but for me the data wasn't stored properly on click)

### CODE REVIEW

**Comment by sanlil** [comment]
I realized now that there is an issue with updating the highlighted options when changing the date. For example, if I mark "light" on April 19 and then go back to April 18 it will still keep "light" marked and same on April 17, but on April 16 it is removed. When going in the other direction (from 16 -> 19) the correct options are marked (i.e. only on April 19).

If anyone already knows what the issue is, feel free to add a commit or write a comment.

**Comment by david8611** [comment]
Nice code adjustments and fixes! It works great to change to the isFocused hook when updating the tracking screen and nice with the addition to be able to clear the storage if needed, etc. 

I didn't have the same issues with the marking as you mensioned above but for some reason I had other problems, for example the latest date (today's date) wasn't being marked at all initially when starting the app and perhaps not stored for me when closing and reopening the app. However by changing the ```isSelected``` state variable to a prop in https://github.com/ovulai/ovulai-react-native-app/pull/59/commits/c3a33a6a151a5ec52b54eafe1338a61fc9c23150 seemed to fix each issue for me. Perhaps you (@sanlil) or someone else can test it on IOS? It seems that asynchronous functions aren't good tools to update two different states which are dependent of each other.

File: App/components/InputLane.jsx [hunk 1]
```diff
@@ -109,35 +86,38 @@ function arrayRemove(array, option) {
...
       <InputOption
         key={e.name}
-        option={e.name}
-        laneKey={laneKey}
-        currentDate={currentDate}
+        optionName={e.name}
+        laneColor={laneColor}
         setLaneState={setLaneState}
         laneState={laneState}
+        isSelected={isSelected}
```
Comment (david8611) [inline App/components/InputLane.jsx:120]: Passing in the isSelected variable as a prop and checking it with the laneState variable seemed to be more reliable than checking it asynchronously using the local storage each time laneState is changing.

**Review by david8611 — APPROVED** [review]

**Review by SamuelHarner — APPROVED** [review]
Highlighting seems to be working for me in the iOS simulator. Here are some screenshots demonstrating this:

**Review by jscarlsson — APPROVED** [review]
Works fine on Android for me. The only thing is that when you clear data, it doesn't go back to the "default" tracking settings (I think we had those, right?) but rather just clears everything. But of course we'll need to actually decide what the default should be so it doesn't matter right now.

**Comment by jscarlsson** [comment]
Oh, when you clear data and then look at Tracking Settings the options are still set, but don't show in the Tracking view. If you disable and then renable they show up again. So I guess we just need to clear those as well.

File: App/screens/TrackingSettings.jsx [hunk 2]
```diff
@@ -102,12 +107,19 @@ const storeSetting = async (key, value) => {
...
+  isEnabled,
+  setIsEnabled
+) => {
   try {
     const keyString = JSON.stringify(key);
     const value = JSON.parse(await AsyncStorage.getItem(keyString));
     if (value === !isEnabled) {
       toggleSwitch();
+    } else if (value === null) {
+      setIsEnabled(false);
```
Comment (david8611) [inline App/screens/TrackingSettings.jsx:122]: This should hopefully fix the issue @jscarlsson mensioned above.

**Comment by sanlil** [comment]
Amazing teamwork!! 🎉 🌟 😍 
Thanks @david8611 for the contributions, @SamuelHarner for the verification and @jscarlsson for the bug finding!

**Pull request merged**